File: pinyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResult;
import entity.Result;

@RestController
@RequestMapping("/brand")
public class BrandController {
	
	@Reference
	private BrandService brandService;
	
	/**
	 * 查询所有品牌
	 * @return 
	 */
	@RequestMapping("/findAll")
	public List<TbBrand> findAll(){
		
		return brandService.findAll();
	}
	
	/**
	 * 分页展示品牌
	 * @param page 当前页码
	 * @param size 每页展示的条数
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult<TbBrand> findPage(int page, int size){
		
		return brandService.findPage(page, size);
	}
	
	/**
	 * 
	 * @param brand
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbBrand brand){
		
		try {
			brandService.add(brand);
			return new Result(true, "添加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "添加失败");
		}
		
	}
	
	
	/**
	 * 通过id查找品牌
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbBrand findOne(Long id){
		
		return brandService.findOne(id);
	}
	
	/**
	 * 修改品牌
	 * @param brand
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody TbBrand brand){
		try {
			brandService.update(brand);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}
	
	/**
	 * 删除品牌
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(Long[] ids){
		try {
			brandService.delete(ids);
			return new Result(true, "删除成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
	
	/**
	 * 条件分页查询
	 * @param brand
	 * @param page
	 * @param size
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult<TbBrand> search(@RequestBody TbBrand brand, int page, int size){
		
		
		return brandService.findPage(brand, page, size);
	}
	
	
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
// 创建品牌控制器
    	app.controller("brandController",function($scope,brandService,$controller){
    		// 继承baseController
    		$controller("baseController",{$scope:$scope});
    		
    		// 查询所有品牌
    		$scope.findAll=function(){
    			brandService.findAll().success(
    				function(response){
    					$scope.list=response;
    			});
       		};
       		
       		
       		
       		/* // 请求后端分页数据(已被search方法代替)
       		$scope.findPage=function(page, size){
       			$http.get("../brand/findPage.do?page="+page+"&size="+size).success(
       				function(pageResult){
       					$scope.brandList=pageResult.rows;
       					$scope.paginationConf.totalItems=pageResult.total;
       				}		
       			
       			);
       			
       		}  */
       		
       		// 添加品牌和修改品牌
       		$scope.save=function(){
       			var object = null;
       			if($scope.entity.id!=null){
       				object=brandService.update($scope.entity);
       			}else{
       				object=brandService.add($scope.entity);
       			}
       			object.success(
       				function(response){
       					if(response.success){
       						$scope.reloadList();
       					}else{
       						alert(response.message);
       					}
       				}		
       			
       			);
       		}
       		
       		// 通过id查找用户
       		$scope.findOne=function(id){
       			brandService.findOne(id).success(
       				function(response){
       					$scope.brand=response;
       				}		
       			)
       			
       		}
       		
       		
       		// 定义一个id数组
       		$scope.selectIds=[];
       		// 批量删除
    		$scope.dele=function(){
    			brandService.dele($scope.selectIds).success(
       				function(response){
       					if(response.success){
       						$scope.reloadList();
       						$scope.selectIds=[];
       					}else{
       						alert(response.message);
       					}
       				}		
       			);
       			
       		}
       		
       		// 定义搜索对象
       		$scope.searchEntity={};
       		
       		// 定义搜索方法
       		$scope.search=function(page,size){
       			brandService.search(page, size, $scope.searchEntity).success(
       				function(response){
       					$scope.list=response.rows;
       					$scope.paginationConf.totalItems=response.total;
       				}		
       			
       			)
       		}
    		
    		
    		
    	});
    